core/src/main/java/de/bwaldvogel/mongo/oplog/CollectionBackedOplog.java

```diff
@@ -85,11 +86,12 @@ public void handleDropCollection(String namespace) {
                 BsonTimestamp timestamp2 = getOplogTimestamp(o2);
                 return timestamp1.compareTo(timestamp2);
             })
-            .map(document -> toChangeStreamResponseDocument(document, changeStreamDocument));
+            .map(document -> toChangeStreamResponseDocument(document, changeStreamDocument)))
+            .stream();
```

Comment:
I would suggest that you extract a method such as `Aggregation.runStagesAsStream(…)` which returns a stream.

Otherwise we work on the stream, collect it into list, and then call `.stream()` on the list again…

Reply:
done